src/ssl_stubs.c

```diff
@@ -434,6 +401,7 @@ CAMLprim value ocaml_ssl_create_context(value protocol, value type)
     caml_acquire_runtime_system();
     caml_raise_constant(*caml_named_value("ssl_exn_context_error"));
   }
+  set_protocol(ctx, Int_val(protocol));
   /* In non-blocking mode, accept a buffer with a different address on
      a write retry (since the GC may need to move it). In blocking
      mode, hide SSL_ERROR_WANT_(READ|WRITE) from us. */
```

Comment:
This comment (which is not new in this PR) make me wonder if we should not have a way to decide if we want SSL_MODE_AUTO_RETRY ? This forces if to be always there, which is not what we want with non blocking socket.

Reply:
After reading the doc, this seems wrong to force SSL_MODE_AUTO_RETRY in all cases. I created issue #122, that you can discuss (there are several fix to this).